### DISCUSSION TITLE
zh/guides/yolo-common-issues/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#11739] [Docs]
# zh/guides/yolo-common-issues/

在Ultralytics 生态系统中使用YOLOv8 时遇到的常见问题的全面故障排除指南。

https://docs.ultralytics.com/zh/guides/yolo-common-issues/

**Comment by yaoshisa** [comment]
Why is the progress bar not displayed when training on the GPU？

**Reply by pderrenger** [reply]
@yaoshisa hi there! It sounds like the progress bar issue might be related to the environment setup or an oversight in the process. Sometimes, when training on a GPU, if the `verbose` flag isn't set correctly, it could prevent the progress bar from displaying.

Here’s a quick check you can do:

```python
from ultralytics import YOLO

# Ensure verbose is set to True
model = YOLO('yolov8n.pt', verbose=True)
results = model.train(data='coco128.yaml')
```

Setting `verbose=True` ensures that progress logs are displayed. If this doesn’t resolve the issue, it could be helpful to check whether your environment supports progress bar visualization or if other output settings might be interfering.

Let me know if this helps or if you need further assistance! 😊

**Comment by echo385** [comment]
Hello, thank you for your contribution! I use the 1280 of imgsz yolov8 to train my 640 images, but my gpu always is stuck ,why？this happens on A100 and 4090 all the time.

**Reply by pderrenger** [reply]
Hello! It sounds like you're experiencing GPU hangs when training with a larger `imgsz` than your original image sizes. This might be due to several reasons:

1. **Memory Overload**: The increase in `imgsz` significantly raises GPU memory requirements. Even high-end GPUs like A100 and 4090 might struggle if the configuration exceeds their memory capacity.

2. **Batch Size**: If you're using a larger `imgsz`, consider lowering the batch size to reduce memory demands.

Here’s what you can try:
- **Reduce the `imgsz`** to match your actual image dimensions or slightly increase it, but not as much as 1280 if it's not necessary.
- **Decrease the batch size** to ensure it fits within your GPU's memory limits.
- **Monitor GPU usage**: Use tools like `nvidia-smi` to watch the memory usage and ensure it's within limits.

If you need further assistance adjusting these settings, feel free to share more details about your setup and training configuration. 🛠️

**Comment by yileicc** [comment]
The bus.jpg has been placed under the assets path during the run, but the " FileNotFoundError: ultralytics/assets/bus.jpg does not exist”

**Reply by glenn-jocher** [reply]
Hello!

Thank you for reaching out. It seems like you're encountering a `FileNotFoundError` for `ultralytics/assets/bus.jpg`. Let's work through this together.

1. **Verify File Path**: Ensure that the `bus.jpg` file is indeed located in the `ultralytics/assets/` directory. You can do this by navigating to the directory and checking if the file exists.

2. **Check Code Implementation**: If you are running a script, make sure the path to the image is correctly specified. Here’s an example of how to load and use the image in a script:
    ```python
    from ultralytics import YOLO

    # Load a pre-trained YOLOv8 model
    model = YOLO("yolov8n.pt")

    # Specify the source image path
    source = "ultralytics/assets/bus.jpg"

    # Make predictions
    results = model.predict(source, save=True, imgsz=320, conf=0.5)
    ```

3. **Update Packages**: Ensure you are using the latest versions of `torch` and `ultralytics`. You can update them using:
    ```bash
    pip install --upgrade torch ultralytics
    ```

4. **Reproducible Example**: If the issue persists, please provide a minimum reproducible example of your code. This will help us better understand the context and provide a more accurate solution. You can find guidance on creating a minimum reproducible example [here](https://docs.ultralytics.com/help/minimum_reproducible_example).

By following these steps, we should be able to pinpoint and resolve the issue. If you have any further questions or need additional assistance, feel free to ask!

**Comment by sjocnjfjd** [comment]
100 epochs completed in 0.146 hours.
Optimizer stripped from runs/detect/train/weights/last.pt, 5.5MB
Optimizer stripped from runs/detect/train/weights/best.pt, 5.5MB

Validating runs/detect/train/weights/best.pt...
Ultralytics 8.3.44 🚀 Python-3.10.12 torch-2.5.1+cu121 CUDA:0 (Tesla T4, 15102MiB)
YOLO11n summary (fused): 238 layers, 2,582,347 parameters, 0 gradients, 6.3 GFLOPs
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
<ipython-input-2-f998661abd0c> in <cell line: 4>()
      2 
      3 model = YOLO('https://hub.ultralytics.com/models/kYDnuTY9JD3hHyFdnc5T')
----> 4 results = model.train(data='"D://python//新建文件夹 (2)//hakusenninnshiki.v2i.yolov11//lane.yaml"', project='runs/train', name='exp', epochs=100)

6 frames
/usr/local/lib/python3.10/dist-packages/ultralytics/data/utils.py in check_det_dataset(dataset, autodownload)
    327             else:
    328                 m += f"\nNote dataset download directory is '{DATASETS_DIR}'. You can update this in '{SETTINGS_FILE}'"
--> 329                 raise FileNotFoundError(m)
    330             t = time.time()
    331             r = None  # success

FileNotFoundError: 
Dataset '/content/datasets/lane/lane.yaml' images not found ⚠️, missing path '/content/datasets/D:python新/建文件夹 (2)/hakusenninnshiki.v2i.yolov11/lane.yaml/images/val'
Note dataset download directory is '/content/datasets'. You can update this in '/root/.config/Ultralytics/settings.json'
why can i deploy

**Reply by glenn-jocher** [reply]
@sjocnjfjd the error indicates a file path issue. Ensure your `data` parameter points to the correct `yaml` file and check that the dataset paths defined in the `yaml` file are valid and accessible. For guidance on dataset preparation and troubleshooting, refer to [YOLO Common Issues](https://docs.ultralytics.com/guides/yolo-common-issues/). Let us know if you need further clarification!

**Comment by sjocnjfjd** [comment]
it is my.yaml：
path: D:python新建文件夹 (2)/hakusenninnshiki.v2i.yolov11/lane.yaml
train: images/train
val: images/val
test: images/test


nc: 1
names: ['hakusen']


roboflow:
&nbsp; workspace: kitazonoken
&nbsp; project: hakusenninnshiki
&nbsp; version: 2
&nbsp; license: CC BY 4.0
&nbsp; url: https://universe.roboflow.com/kitazonoken/hakusenninnshiki/dataset/2
is it right ?



---Original---
From: "Glenn ***@***.***&gt;
Date: Fri, Dec 6, 2024 23:59 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [ultralytics/ultralytics] zh/guides/yolo-common-issues/(Discussion #11739)




 
@sjocnjfjd the error indicates a file path issue. Ensure your data parameter points to the correct yaml file and check that the dataset paths defined in the yaml file are valid and accessible. For guidance on dataset preparation and troubleshooting, refer to YOLO Common Issues. Let us know if you need further clarification!

**Reply by glenn-jocher** [reply]
@sjocnjfjd your YAML file seems mostly correct, but be cautious with the file path formatting. Replace `D:python新建文件夹 (2)/hakusenninnshiki.v2i.yolov11/lane.yaml` with a properly formatted path (e.g., `D:/python/新建文件夹 (2)/hakusenninnshiki.v2i.yolov11/lane.yaml`). Ensure all dataset paths are accessible and the Roboflow `url` is valid. If issues persist, refer to the [YOLO Common Issues guide](https://docs.ultralytics.com/guides/yolo-common-issues/) for troubleshooting.

**Comment by sjocnjfjd** [comment]
I have looled&nbsp; the issue&nbsp; guides,but I can not find what I&nbsp; want to solve my problem

---Original---
From: "Glenn ***@***.***&gt;
Date: Sat, Dec 7, 2024 01:27 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [ultralytics/ultralytics] zh/guides/yolo-common-issues/(Discussion #11739)




 
@sjocnjfjd your YAML file seems mostly correct, but be cautious with the file path formatting. Replace D:python新建文件夹 (2)/hakusenninnshiki.v2i.yolov11/lane.yaml with a properly formatted path (e.g., D:/python/新建文件夹 (2)/hakusenninnshiki.v2i.yolov11/lane.yaml). Ensure all dataset paths are accessible and the Roboflow url is valid. If issues persist, refer to the YOLO Common Issues guide for troubleshooting.

**Reply by glenn-jocher** [reply]
@sjocnjfjd thank you for your comment. If you've already reviewed the issue guides and couldn't find a solution, please provide specific details about your problem, such as the error message or unexpected behavior you're encountering. This will help us assist you more effectively. Additionally, you can reference the [YOLO Common Issues guide](https://docs.ultralytics.com/guides/yolo-common-issues/) for further troubleshooting. Let us know how we can assist!

**Comment by Mz-Han** [comment]
Hello, I am using the trained yolov26s-seg.pt for small object segmentation. When I set max-det to a value greater than 500, an error occurs. Would you have time to help me with this? Thank you very much！

torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate4.95GiB (GPU 0;23.99 GiB total capacity;17.39 GiB already allocated; 0 bytes free; 17.65 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF

**Reply by glenn-jocher** [reply]
That error is expected when `max_det` is pushed very high for segmentation—memory scales with the number of kept detections (and masks), so `max_det>500` can OOM on a 24GB GPU; best next steps are to raise `conf`, lower `imgsz`, or keep `max_det` lower (see the GPU/OOM notes in [YOLO common issues troubleshooting](https://docs.ultralytics.com/zh/guides/yolo-common-issues/)). If you think it’s a bug, please paste a text (not screenshot) minimal repro command plus `yolo checks` output as described in [Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum-reproducible-example/).

```bash
yolo task=segment mode=predict model=yolo26s-seg.pt source=your.mp4 imgsz=640 conf=0.4 max_det=300 half=True device=0
yolo checks
```

If reserved memory is much larger than allocated, also try setting this **before** running Python (then restart your session/kernel):

```bash
export PYTORCH_CUDA_ALLOC_CONF=max_split_size_mb:128
```
